// jsonresume-theme-projects/dist/index.cjs
"use strict";Object.defineProperty(exports,Symbol.toStringTag,{value:"Module"});const y=require("micromark"),k=require("striptags"),b=require("feather-icons");function e(t,...i){return t.reduce((r,o,a)=>{const n=i[a];return Array.isArray(n)?r+o+n.join(""):n!=null&&!!n!==n?r+o+n:r+o},"")}function s(t,i=!1){const r=y(t);return i?k(r):r}const x=t=>new Date(t).toLocaleDateString("en",{month:"short",year:"numeric",timeZone:"UTC"});function u(t){return e`<time datetime="${t}">${x(t)}</time>`}function w(t=[]){return t.length>0&&e`
    <section id="awards">
      <h3>Awards</h3>
      <div class="stack">
        ${t.map(({awarder:i,date:r,summary:o,title:a})=>e`
          <article>
            <header>
              <h4>${a}</h4>
              <div class="meta">
                ${i&&e`
                  <div>
                    Awarded by <strong>${i}</strong>
                  </div>
                `}
                ${r&&u(r)}
              </div>
            </header>
            ${o&&s(o)}
          </article>
        `)}
      </div>
    </section>
  `}const L=t=>t.replace(/^(https?:|)\/\//,"").replace(/\/$/,"");function h(t,i){return i?t?e`<a href="${t}">${i}</a>`:i:t&&e`<a href="${t}">${L(t)}</a>`}function j(t=[]){return t.length>0&&e`
    <section id="certificates">
      <h3>Certificates</h3>
      <div class="stack">
        ${t.map(({date:i,issuer:r,name:o,url:a})=>e`
          <article>
            <header>
              <h4>${h(a,o)}</h4>
              <div class="meta">
                ${r&&e`
                  <div>
                    Issued by <strong>${r}</strong>
                  </div>
                `}
                ${i&&u(i)}
              </div>
            </header>
          </article>
        `)}
      </div>
    </section>
  `}function p(t,i){return e`${u(t)} – ${i?u(i):"Present"}`}function z(t=[]){return t.length>0&&e`
    <section id="education">
      <h3>Education</h3>
      <div class="stack">
        ${t.map(({area:i,courses:r=[],institution:o,startDate:a,endDate:n,studyType:l,url:c})=>e`
          <article>
            <header>
              <h4>${h(c,o)}</h4>
              <div class="meta">
                ${i&&e`<strong>${i}</strong>`}
                <div>${p(a,n)}</div>
              </div>
            </header>
            ${l&&s(l)}
            ${r.length>0&&e`
              <h5>Courses</h5>
              <ul>
                ${r.map(d=>e`<li>${s(d)}</li>`)}
              </ul>
            `}
          </article>
        `)}
      </div>
    </section>
  `}function g(t,i){return(b.icons[t.toLowerCase()]||b.icons[i.toLowerCase()]).toSvg({width:16,height:16})}const C=t=>Intl.DisplayNames?new Intl.DisplayNames(["en"],{type:"region"}).of(t):t;function I(t={}){const{email:i,image:r,label:o,location:a,name:n,phone:l,profiles:c=[],summary:d,url:$}=t;return e`
    <header class="masthead">
      ${r&&e`<img src="${r}" alt="">`}
      <div>
        ${n&&e`<h1>${n}</h1>`}
        ${o&&e`<h2>${o}</h2>`}
      </div>
      ${d&&e`<article>${s(d)}</article>`}
      <ul class="icon-list">
        ${a?.city&&e`
          <li>
            ${g("map-pin")}
            ${a.city}${a.countryCode&&e`, ${C(a.countryCode)}`}
          </li>
        `}
        ${i&&e`
          <li>
            ${g("mail")}
            <a href="mailto:${i}">${i}</a>
          </li>
        `}
        ${l&&e`
          <li>
            ${g("phone")}
            <a href="tel:${l.replace(/\s/g,"")}">${l}</a>
          </li>
        `}
        ${$&&e`
          <li>
            ${g("link")}
            ${h($)}
          </li>
        `}
        ${c.map(({network:v,url:m,username:f})=>e`
          <li>
            ${g(v,"user")}
            ${h(m,f)}
            ${v&&e`<span class="network">(${v})</span>`}
          </li>
        `)}
      </ul>
    </header>
  `}function q(t=[]){return t.length>0&&e`
    <section id="interests">
      <h3>Interests</h3>
      <div class="grid-list">
        ${t.map(({keywords:i=[],name:r})=>e`
          <div>
            ${r&&e`<h4>${r}</h4>`}
            ${i.length>0&&e`
              <ul class="tag-list">
                ${i.map(o=>e`<li>${o}</li>`)}
              </ul>
            `}
          </div>
        `)}
      </div>
    </section>
  `}function P(t=[]){return t.length>0&&e`
    <section id="languages">
      <h3>Languages</h3>
      <div class="grid-list">
        ${t.map(({fluency:i,language:r})=>e`
          <div>
            ${r&&e`<h4>${r}</h4>`}
            ${i}
          </div>
        `)}
      </div>
    </section>
  `}function R(t={}){const{name:i,summary:r}=t;return e`
    ${i&&e`<title>${i}</title>`}
    ${r&&e`<meta name="description" content="${s(r,!0)}">`}
  `}const S=t=>Intl.ListFormat?new Intl.ListFormat("en").format(t):t.join(", ");function A(t=[]){return t.length>0&&e`
    <section id="projects">
      <h3>Projects</h3>
      <div class="stack">
      ${t.map(({description:i,entity:r,highlights:o=[],keywords:a=[],name:n,startDate:l,endDate:c,roles:d=[],type:$,url:v})=>e`
        <article>
          <header>
            <h4>${h(v,n)}</h4>
            <div class="meta">
              <div>
                ${d.length>0&&e`<strong>${S(d)}</strong>`}
                ${r&&e`at <strong>${r}</strong>`}
              </div>
              <div>${p(l,c)}</div>
              ${$&&e`<div>${$}</div>`}
            </div>
          </header>
          ${i&&s(i)}
          ${o.length>0&&e`
            <ul>
              ${o.map(m=>e`<li>${s(m)}</li>`)}
            </ul>
          `}
          ${a.length>0&&e`
            <ul class="tag-list">
              ${a.map(m=>e`<li>${m}</li>`)}
            </ul>
          `}
        </article>
      `)}
      </div>
    </section>
  `}function D(t=[]){return t.length>0&&e`
    <section id="publications">
      <h3>Publications</h3>
      <div class="stack">
        ${t.map(({name:i,publisher:r,releaseDate:o,summary:a,url:n})=>e`
          <article>
            <header>
              <h4>${h(n,i)}</h4>
              <div class="meta">
                ${r&&e`
                  <div>
                    Published by <strong>${r}</strong>
                  </div>
                `}
                ${o&&u(o)}
              </div>
            </header>
            ${a&&s(a)}
          </article>
        `)}
      </div>
    </section>
  `}function O(t=[]){return t.length>0&&e`
    <section id="references">
      <h3>References</h3>
      <div class="stack">
        ${t.map(({name:i,reference:r})=>e`
          <blockquote>
            ${r&&s(r)}
            ${i&&e`
              <p>
                <cite>${i}</cite>
              </p>
            `}
          </blockquote>
        `)}
      </div>
    </section>
  `}function W(t=[]){return t.length>0&&e`
    <section id="skills">
      <h3>Skills</h3>
      <div class="grid-list">
        ${t.map(({keywords:i=[],name:r})=>e`
          <div>
            ${r&&e`<h4>${r}</h4>`}
            ${i.length>0&&e`
              <ul class="tag-list">
                ${i.map(o=>e`<li>${o}</li>`)}
              </ul>
            `}
          </div>
        `)}
      </div>
    </section>
  `}function E(t=[]){return t.length>0&&e`
    <section id="volunteer">
      <h3>Volunteer</h3>
      <div class="stack">
        ${t.map(({highlights:i=[],organization:r,position:o,startDate:a,endDate:n,summary:l,url:c})=>e`
          <article>
            <header>
              <h4>${h(c,r)}</h4>
              <div class="meta">
                <strong>${o}</strong>
                <div>${p(a,n)}</div>
              </div>
            </header>
            ${l&&s(l)}
            ${i.length>0&&e`
              <ul>
                ${i.map(d=>e`<li>${s(d)}</li>`)}
              </ul>
            `}
          </article>
        `)}
      </div>
    </section>
  `}function F(t=[]){const i=t.reduce((r,{description:o,name:a,url:n,...l})=>{const c=r[r.length-1];return c&&c.name===a&&c.description===o&&c.url===n?c.items.push(l):r.push({description:o,name:a,url:n,items:[l]}),r},[]);return t.length>0&&e`
    <section id="work">
      <h3>Work</h3>
      <div class="stack">
        ${i.map(({description:r,name:o,url:a,items:n=[]})=>e`
          <article>
            <header>
              <h4>${h(a,o)}</h4>
              <div class="meta">
                ${r&&e`<div>${r}</div>`}
              </div>
            </header>
            <div class="timeline">
              ${n.map(({highlights:l=[],location:c,position:d,startDate:$,endDate:v,summary:m})=>e`
                <div>
                  <div>
                    <h5>${d}</h5>
                    <div class="meta">
                      <div>${p($,v)}</div>
                      ${c&&e`<div>${c}</div>`}
                    </div>
                  </div>
                  ${m&&s(m)}
                  ${l.length>0&&e`
                    <ul>
                      ${l.map(f=>e`<li>${s(f)}</li>`)}
                    </ul>
                  `}
                </div>
              `)}
            </div>
          </article>
        `)}
      </div>
    </section>
  `}function M(t={}){const{colors:i}=t;return i&&Object.entries(i).map(([r,[o,a=o]])=>`--color-${r}-light:${o}; --color-${r}-dark:${a};`).join(" ")}function N(t,i){return e`<!doctype html>
    <html lang="en" style="${M(t.meta)}">
      <head>
        <meta charset="utf-8" />
        ${R(t.basics)}
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="stylesheet" href=print.css" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700&display=swap" />
        <style>
          ${i}
        </style>
      </head>
      <body>
        ${I(t.basics)} ${F(t.work)} ${E(t.volunteer)} ${A(t.projects)}
        ${w(t.awards)} ${j(t.certificates)} ${D(t.publications)}
        ${W(t.skills)} ${z(t.education)} ${P(t.languages)}
        ${q(t.interests)} ${O(t.references)}
      </body>
    </html>`}const T=`:root{color-scheme:light dark;--color-background-light: #ffffff;--color-dimmed-light: #f3f4f5;--color-primary-light: #191e23;--color-secondary-light: #6c7781;--color-accent-light: #0073aa;--color-background-dark: #191e23;--color-dimmed-dark: #23282d;--color-primary-dark: #fbfbfc;--color-secondary-dark: #ccd0d4;--color-accent-dark: #00a0d2;--color-background: var(--color-background-light);--color-dimmed: var(--color-dimmed-light);--color-primary: var(--color-primary-light);--color-secondary: var(--color-secondary-light);--color-accent: var(--color-accent-light);--scale-ratio: 1.25;--scale0: 1rem;--scale1: calc(var(--scale0) * var(--scale-ratio));--scale2: calc(var(--scale1) * var(--scale-ratio));--scale3: calc(var(--scale2) * var(--scale-ratio));--scale4: calc(var(--scale3) * var(--scale-ratio));--scale5: calc(var(--scale4) * var(--scale-ratio))}@media (prefers-color-scheme: dark){:root{--color-background: var(--color-background-dark);--color-dimmed: var(--color-dimmed-dark);--color-primary: var(--color-primary-dark);--color-secondary: var(--color-secondary-dark);--color-accent: var(--color-accent-dark)}}*{box-sizing:border-box;margin:0;padding:0}html{font-size:14px}body{background:var(--color-background);color:var(--color-primary);display:grid;font:1em/1.5 Lato,sans-serif;gap:2em;grid-template-columns:[full-start] 1fr [main-start side-start] minmax(min-content,12em) [side-end content-start] minmax(min-content,36em) [main-end content-end] 1fr [full-end];grid-template-rows:auto [content] 0;margin-bottom:4em}body:before{content:"";grid-column:full;grid-row:content}ol,ul{padding-left:1em}:not(.icon-list,.tag-list)>li+li{margin-top:.4em}li::marker,.network{color:var(--color-secondary)}a{color:var(--color-accent);text-decoration:none}a:focus,a:hover{text-decoration:underline}h1,h2,h3,h5{font-weight:400}h1,h2,h3{line-height:1.2}h1{font-size:var(--scale5)}h2{color:var(--color-secondary);font-size:var(--scale4)}h3{color:var(--color-secondary);font-size:var(--scale3);grid-column:side;margin-bottom:1rem}h4{font-size:var(--scale2)}h5{font-size:var(--scale1)}h6{font-size:var(--scale0)}blockquote{border-left:.2em solid var(--color-dimmed);padding-left:1em}cite{color:var(--color-secondary);font-style:inherit}cite:before{content:"— "}svg{margin-right:.2em;vertical-align:text-bottom}.masthead{background:var(--color-dimmed);display:inherit;gap:inherit;grid-column:full;grid-template-columns:inherit;padding:4em 0 1em;text-align:center}.masthead>*,section{grid-column:main}.masthead>img{border:4px solid;border-radius:50%;margin:0 auto;max-width:12em}article>*+*,blockquote>*+*,.timeline>div>*+*{margin-top:.6em}.meta{color:var(--color-secondary)}.stack{display:grid;gap:1.5em}.icon-list{display:flex;flex-wrap:wrap;gap:.4em 1em;justify-content:center;list-style:none;padding:0}.grid-list{display:grid;gap:1em}.tag-list{display:flex;flex-wrap:wrap;gap:.4em;list-style:none;padding:0}.tag-list>li{background:var(--color-dimmed);border-radius:.2em;padding:.2em .6em}.timeline>div{position:relative}.timeline>div:not(:last-child){padding-bottom:1rem}.timeline>div:not(:last-child):before{content:"";position:absolute;top:1rem;left:-15px;width:2px;height:100%;background:var(--color-secondary)}.timeline>div:not(:only-child):after{content:"";position:absolute;top:.6rem;left:-20px;width:8px;height:8px;background:var(--color-secondary);border:2px solid var(--color-background);border-radius:50%}@media print,(min-width: 48em){h3{text-align:right;margin-bottom:inherit}.masthead{text-align:inherit}.masthead>*,section{grid-column:content}.masthead img{grid-column:side;grid-row:span 2;max-width:100%}section{display:contents}.icon-list{flex-direction:column}.grid-list{grid-template-columns:repeat(auto-fit,minmax(calc((100% - 1em)/2),1fr))}}
`,U={mediaType:"print"},V=t=>N(t,T);exports.pdfRenderOptions=U;exports.render=V;
